### PR TITLE
Add test for vacuum with CDF files

### PR DESCRIPTION
**Pull request by homar**
## Description
Depends on https://github.com/trinodb/trino/pull/16205
Add test making sure vacuum doesn't delete cdf files that are not old enough for cleanup

## Additional context and related issues
Previous implementation deleted all CDF files no matter their age.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
() Release notes are required, with the following suggested text:

### CODE REVIEW

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeTransactionLogEntry.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ public static DeltaLakeTransactionLogEntry fromJson(
             @JsonProperty("metaData") MetadataEntry metaData,
             @JsonProperty("protocol") ProtocolEntry protocol,
             @JsonProperty("commitInfo") CommitInfoEntry commitInfo,
-            @JsonProperty("cdfFileEntry") CDFFileEntry cdfFileEntry)
+            @JsonProperty("cdc") CDFFileEntry cdfFileEntry)
```
Comment (alexjo2144) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeTransactionLogEntry.java:59]: Why?

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java [hunk 2]
```diff
@@ -101,7 +101,8 @@
         REMOVE("remove"),
         METADATA("metadata"),
         PROTOCOL("protocol"),
-        COMMIT("commitinfo");
+        COMMIT("commitinfo"),
+        CDC("cdc");
```
Comment (alexjo2144) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java:105]: I thought you said checkpoint files don't contain CDC information

**Review by alexjo2144 — CHANGES_REQUESTED** [review]
Only reviewing last commit.

If my memory is right that checkpoint files don't contain CDC references, this is not going to work. If the most recent commit was checkpointed `getJsonTransactionLogEntries` is going to return an empty list and you'll miss some valid CDC files.

I would argue that what we have now is good enough, comparing the CDC file timestamp against the cutoff. It doesn't ensure that we keep the CDC files for the most recent checkpoint, but I don't think that's a big deal.

**Comment by bitsondatadev** [comment]
@homar, is this still being actively worked on?

**Comment by homar** [comment]
> @homar, is this still being actively worked on?

Not actively, but I will work on this soon, this or next week.

**Comment by bitsondatadev** [comment]
Sounds good :) I'll leave it alone!

**Comment by alexjo2144** [comment]
@homar do you want to convert this to a Databricks compatibility test so that it doesn't depend on you other PR?
Or just include it in the table_changes PR?

**Comment by homar** [comment]
I will include it in table_changes

**Comment by findepi** [comment]
> I will include it in [table_changes](https://github.com/trinodb/trino/pull/16205)

should this one be closed now?

**Pull request closed without merging**